Persephone/Controllers/WindowController.swift

```diff
@@ -20,6 +20,11 @@ class WindowController: NSWindowController {
     super.windowDidLoad()
     window?.titleVisibility = .hidden
 
+    NSEvent.addLocalMonitorForEvents(matching: .keyDown) {
```

Comment:
The documentation suggests that the return value from this method should be saved and later passed to `removeMonitor(_:)`.

Reply:
Curious if there's a way to get space bar presses via the responder chain, without needing to add the extra monitor?

Reply:
I don't entirely understand the responder chain yet. It seemed like the spacebar key was getting caught by the queue outline view and not passed on to the window. I figured the window would be the first in the chain?

Reply:
Events on the responder chain start with the first responder, and then walk up the responder chain until the first object handles the event. Let’s say you had a search box — this would be how you’d have the space bar type a space character while the search field is first responder, until you’re done and then something higher up the chain gets a chance to handle it.

Reply:
Oh I see, so with this event monitor the search box would never get a space?

Do I need to configure the queue outline view to somehow ignore the space bar and pass it up the chain?

Reply:
My guess is that with the current implementation, _both_ the search box and the window controller would handle the space, which is probably not what you’d want. I think if you returned `nil` from the monitor, the search box might not get the event at all.

What action does the space bar perform when the outline view is first responder? Does the outline view always get key presses even after you first launch the app? Do you think it would make sense to always have space bar go to the window, and have the outline view just ignore it?

If space bar does something useful in the outline view, it might make sense to avoid overriding that default. But it might totally make sense for this app. (Side note: is an outline view the same thing as the Finder list view? If so, Finder doesn’t seem to do anything when space is pressed.)

If you want to override `NSOutlineView`’s default handling, I think you could override its `keyDown(with:)` method something like this:

```swift
override func keyDown(with event: NSEvent) {
  switch event.keyCode {
  case NSEvent.keyCodeSpace:
    nextResponder.keyCode(with: event)
  default:
    super.keyCode(with: event)
  }
}
```

(Based on [Event Handling Basics](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/EventOverview/EventHandlingBasics/EventHandlingBasics.html).)

Reply:
Ok, I've switched this over to use the responder chain. This involved creating custom classes for the split view and the window to pass the events along as the chain looks something like this:

QueueView → SplitViewController → Window → WindowController

Reply:
Nice, although it’s a bummer to have to manually pass these along at each step. Do they all really consume the key press without this?

Reply:
Yeah, the keypress does not get through.